src/data/data.js

```diff
@@ -171,10 +173,93 @@ extend(ChartInternal.prototype, {
 		return typeof x !== "undefined" ? x : null;
 	},
 
-	getMaxDataCount() {
+	/**
+	 * Get min/max value from the data
+	 * @private
+	 * @param {Array} data array data to be evaluated
+	 * @return {{min: {Number}, max: {Number}}}
+	 */
+	getMinMaxValue(data) {
+		let min;
+		let max;
+
+		(data || this.data.targets.map(t => t.values))
+			.forEach(v => {
+				min = d3Min([min, d3Min(v, t => t.value)]);
+				max = d3Max([max, d3Max(v, t => t.value)]);
+			});
+
+		return {min, max};
+	},
+
+	/**
+	 * Get the min/max data
+	 * @private
+	 * @return {{min: Array, max: Array}}
+	 */
+	getMinMaxData() {
+		const data = this.data.targets.map(t => t.values);
+		const minMax = this.getMinMaxValue(data);
+
+		let min = [];
+		let max = [];
+
+		data.forEach(v => {
+			const minData = this.getFilteredDataByValue(v, minMax.min);
+			const maxData = this.getFilteredDataByValue(v, minMax.max);
+
+			if (minData.length) {
+				min = min.concat(minData);
+			}
+
+			if (maxData.length) {
+				max = max.concat(maxData);
+			}
+		});
+
+		return {min, max};
+	},
+
+	/**
+	 * Get filtered data by value
+	 * @private
+	 * @param {Object} data
+	 * @param {Number} value
+	 * @return {Array} filtered array data
+	 */
+	getFilteredDataByValue(data, value) {
 		const $$ = this;
 
-		return d3Max($$.data.targets, t => t.values.length);
+		return data.filter(t => {
+			if (t.value === value) {
+				// select element. It needs schedule by setTimeout to avoid collision with d3's transitions tween
+				setTimeout(() => (t.element = $$.getElementByDataIndex(t)), 0);
```

Comment:
Why do you handle asynchronous operations synchronously?
How about handling asynchronous operations with callbacks or promises?

Reply:
I wrote the reason on comment.
During the initialization process, chart is rendered with transition and during the transition if some d3 related calls are called, it cause `maximum call stack size exceeded` error. 
Is to avoid that situation.

Reply:
ok. I just wanted the asynchronous operation to be explicitly revealed.